apps/crud/src/app/app.component.ts

```diff
@@ -1,52 +1,185 @@
+  todo: Todo;
+  callState: CallState;
+};
 
 @Component({
+  selector: 'todo-item',
+  imports: [
+    CommonModule,
+    LetModule,
+    MatProgressSpinnerModule,
+    MatButtonModule,
+    MatSnackBarModule,
+  ],
   standalone: true,
-  imports: [CommonModule],
-  selector: 'app-root',
   template: `
-    <div *ngFor="let todo of todos">
-      {{ todo.title }}
-      <button (click)="update(todo)">Update</button>
-    </div>
+    <ng-container *ngIf="vm$ | async as vm">
+      <div *ngIf="vm?.error">
+        <p>{{ vm?.error?.message }}</p>
+        <button mat-button color="warn" (click)="deleteTodo$(vm?.todo)">
+          Retry
+        </button>
+      </div>
+
+      <mat-spinner *ngIf="vm?.callState === 'LOADING'" [diameter]="40">
+      </mat-spinner>
+
+      <ng-container *ngIf="vm?.callState === 'LOADED'">
+        <div *ngrxLet="vm.todo as todo">
+          {{ todo.title }}
+          <button
+            mat-stroked-button
+            color="primary"
+            (click)="updateTodo$(todo)">
+            Update
+          </button>
+          <button mat-stroked-button color="warn" (click)="deleteTodo$(todo)">
+            Delete
+          </button>
+        </div>
+      </ng-container>
+    </ng-container>
   `,
-  styles: [],
+  providers: [ComponentStore],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styles: [
+    `
+      :host {
+        padding: 20px;
+        display: flex;
+        gap: 10px;
+      }
+    `,
+  ],
 })
-export class AppComponent implements OnInit {
-  todos!: any[];
-
-  constructor(private http: HttpClient) {}
-
-  ngOnInit(): void {
-    this.http
-      .get<any[]>('https://jsonplaceholder.typicode.com/todos')
-      .subscribe((todos) => {
-        console.log('return', todos);
-        this.todos = todos;
-      });
+export class TodoItemComponent extends ComponentStore<TodoItem> {
+  readonly vm$ = this.select({
+    todo: this.select((state) => state.todo),
+    callState: this.select((state) => state.callState),
+    error: this.select(({ callState }) => {
+      if ((callState as ErrorState).error !== undefined) {
+        return (callState as ErrorState).error;
+      }
+      return null;
+    }),
+  }).pipe(tap(console.log));
+
+  constructor(private todoService: TodoService, private snackBar: MatSnackBar) {
+    super({
+      todo: {} as Todo,
+      callState: 'INIT',
+    });
   }
 
-  update(todo: any) {
-    this.http
-      .put<any>(
-        `https://jsonplaceholder.typicode.com/todos/${todo.id}`,
-        JSON.stringify({
-          todo: todo.id,
-          title: randText(),
-          body: todo.body,
-          userId: todo.userId,
-        }),
-        {
-          headers: {
-            'Content-type': 'application/json; charset=UTF-8',
-          },
-        }
+  @Input()
+  set todoItem(todo: Todo) {
+    this.patchState({
+      todo,
+      callState: 'LOADED',
+    });
+  }
+
+  readonly updateTodo$ = this.effect((todo: Observable<Todo>) => {
```

Comment:
just to be picky, this is a function, not an observable, so don't use the $ sign